### PR TITLE
Adding command to validate pre and post synthesis ports

### PR DESCRIPTION
**Pull request by behzadmehmood-rs**
> ### Motivate of the pull request
> - [ ] To address an existing issue. If so, please provide a link to the issue: <issue id>
> - [x] Breaking new feature. If so, please describe details in the description part.

> ### Describe the technical details
Adding a command to validate pre and post synthesis ports.
> #### What is currently done? (Provide issue link if applicable)
> 
> 
> 
> 
> 
>
Currently, there is no command to check if the ports are same before and after synthesis.

> #### What does this pull request change?
> 
> 
> 
> 
> 
Now the ports of the synthesized netlist will be compared to the RTL ports. In case of a missing port after synthesis, an error will be generated and in case of an addition of new port, a warning will be issued.

> ### Which part of the code base require a change
> 
> - [ ] Library: <Specify the library name>
> - [ ] Plug-in: <Specify the plugin name>
> - [ ] Engine
> - [ ] Documentation
> - [ ] Regression tests
> - [ ] Continous Integration (CI) scripts

> ### Impact of the pull request

> - [ ] Require a change on Quality of Results (QoR)
> - [ ] Break back-compatibility. If so, please list who may be influenced.

### CODE REVIEW

File: src/Compiler/Compiler.cpp [hunk 1]
```diff
@@ -3036,4 +3043,29 @@ int Compiler::add_files(Compiler* compiler, Tcl_Interp* interp, int argc,
...
+      FilePath(Action::Analyze, "port_info.json").string();
+  std::string nlPortInfo =
+      FilePath(Action::Pack, "post_synth_ports.json").string();
+  auto binPath = GlobalSession->Context()->BinaryPath();
+  std::string command = binPath.string() + std::string("/verify_synth_ports ") +
+                        "--rtlPortInfo " + rtlPortInfo + " --nlPortInfo " +
+                        nlPortInfo;
+  int systemRet = std::system(command.c_str());
+  if (systemRet == -1) {
+    compiler->ErrorMessage("Error: Error in verify_synth_ports");
```
Comment (ravic-rs) [inline src/Compiler/Compiler.cpp:3068]: Remove the prepend here.

File: src/Compiler/Compiler.cpp [hunk 2]
```diff
@@ -3036,4 +3043,29 @@ int Compiler::add_files(Compiler* compiler, Tcl_Interp* interp, int argc,
...
+  compiler->Message(out.str());
+  std::string rtlPortInfo =
+      FilePath(Action::Analyze, "port_info.json").string();
+  std::string nlPortInfo =
+      FilePath(Action::Pack, "post_synth_ports.json").string();
+  auto binPath = GlobalSession->Context()->BinaryPath();
+  std::string command = binPath.string() + std::string("/verify_synth_ports ") +
+                        "--rtlPortInfo " + rtlPortInfo + " --nlPortInfo " +
+                        nlPortInfo;
+  int systemRet = std::system(command.c_str());
```
Comment (ravic-rs) [inline src/Compiler/Compiler.cpp:3066]: Please use "ExecuteAndMonitorSystemCommand" function here.

**Review by ravic-rs — CHANGES_REQUESTED** [review]
Atleast 2 comments to review.

**Comment by ravic-rs** [comment]
@behzadmehmood-rs did you get a chance to review the comments ?

**Comment by behzadmehmood-rs** [comment]
> @behzadmehmood-rs did you get a chance to review the comments ?

@ravic-rs  I could not work today due to some health issues, I will make changes tomorrow.

File: src/Compiler/Compiler.cpp (hunk 1 above)
Comment (behzadmehmood-rs) [reply src/Compiler/Compiler.cpp]: Done.

File: src/Compiler/Compiler.cpp (hunk 2 above)
Comment (behzadmehmood-rs) [reply src/Compiler/Compiler.cpp]: Done.

**Review by ravic-rs — APPROVED** [review]

**Pull request merged**